pallets/parachain-staking/src/lib.rs

```diff
@@ -996,25 +959,20 @@ pub mod pallet {
 		}
 
 		#[pallet::weight(
-			<T as Config>::WeightInfo::execute_leave_candidates(*candidate_delegation_count)
-		)]
+        < T as Config >::WeightInfo::execute_leave_candidates(< CandidateInfo < T >>::get(candidate).ok_or(Error::< T >::CandidateDNE).unwrap().delegation_count)
+        )]
```

Comment:
minor issue. But ok_or(Error::<**>::***).unwrap() is not proper.
Consider using [ok_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else)

CandidateMetadata::new() in case of "else".

Reply:
map_or_else